utils/oauth.go

```diff
@@ -0,0 +1,131 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+var ErrClientIdNotFound = errors.New("ERROR: GITHUB OAUTH CLIENT ID NOT FOUND")
+var ErrClientSecretNotFound = errors.New("ERROR: GITHUB OAUTH CLIENT SECRET NOT FOUND")
+var ErrGithubAPIError = errors.New("ERROR: GITHUB API ERROR")
+
+// Body fields for the request
+type OAuthAccessReqFields struct {
+	ClientId     string `json:"client_id"`
+	ClientSecret string `json:"client_secret"`
+	Code         string `json:"code"`
+}
+
+// Body fields for the response
+type OAuthAccessResFields struct {
+	AccessToken string `json:"access_token"`
+	Error       string `json:"error"`
+}
+
+func GetOauthAccessToken(code string) (string, error) {
+	client_id := os.Getenv("GITHUB_OAUTH_CLIENT_ID")
+	client_secret := os.Getenv("GITHUB_OAUTH_CLIENT_SECRET")
+
+	if client_id == "" {
+		return "", ErrClientIdNotFound
+	}
+
+	if client_secret == "" {
+		return "", ErrClientSecretNotFound
+	}
+
```

Comment:
How do you feel about checking these values in the start of the server itself?A function which initiliazes variables and checks them, if faulty doesnt allow the server to start. Can add a dev flag which doesnt check for development feasibility

Reply:
I think it is better this way since environment variables can be set or unset during runtime. Also helps with testing.